Source/gamemenu.cpp

```diff
@@ -134,7 +134,9 @@ void gamemenu_load_game(BOOL bActivate)
 
 void gamemenu_save_game(BOOL bActivate)
 {
-	if (pcurs != CURSOR_HAND) {
+	// Controller support: Allow saving without the cursor
+	// if (pcurs != CURSOR_HAND) {
+	if (pcurs > CURSOR_HAND) {

```

Comment:
Maybe it would make more sense to simply not draw the cursor if the controller is currently active:
https://github.com/diasurgical/devilutionX/blob/master/Source/scrollrt.cpp#L105

This might also make it easier to support switching input mode during game play.

Reply:
The right stick of the controller can be used to move and click the mouse (e.g. for precise spellcasting).

This check is a mysterious check in the original source that doesn't allow you to activate the "Save Game" menu if the cursor is invisible (for no good reason as far as I can tell).

Reply:
Yes, and when that is happening you can enable cursor rendering. Else you will have to hunt down all the places where there is a check for the current cursor, and other logic might make the cursor show if it changes the current cursor sprite.

Reply:
Hmm, that's exactly what I'm doing so I'm not sure what you mean then.

If the cursor changes the sprite, it should show normally, as it usually happens when casting a special skill/spell.

Reply:
I'm suggesting that instead of changing all the places that check for `pcurs`, you add a new global `gbShowCursor` and check it in scrollrt_draw_cursor_item. Then instead of calling `SetCursor_(CURSOR_NONE)` you set `gbShowCursor = false`. It actually looks like this more or less already exists in the form of `newCurHidden`?

Reply:
Then in all the places that check for `pcurs == 1` do we check `pcurs == 1 || pcurs == 0 && !gbShowCursor`?

Reply:
No, they should all work fine as is, because `pcurs` behaves exactly as before. Line 105 in scrollrt.cpp simply has to change to:
`	if (gbHideCursor || pcurs <= 0 || cursW == 0 || cursH == 0) {`
(gbHideCursor is probably better then gbShowCursor, since it should show by default)

Reply:
In case the code is not really handling when the cursor should show then maybe this would work instead:
` if ((gbHideCursor && pcurs == 1) || pcurs <= 0 || cursW == 0 || cursH == 0) {`

Reply:
Would you be interested in doing this? There are only 11 places where I had to change this check in Source (can be found by looking for `// Controller support`).

Reply:
Sure I can look in to it :)

Reply:
I have now rewritten it to use `curHide`, I modified your commits to avoid any churn in Source. So just be aware that you have to hard reset the next time you fetch the branch. Let me know if you need help with rebasing any commits that you might have done offline in the meantime.